src/thunderbots/software/ai/navigator/path_planner/violation_function.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include "geom/point.h"
+
+/**
+ * The ViolationFunction is an object that represents a function
+ * that can be called to evaluate the boundary violation of a
+ * point on the field, or implement other soft avoidance areas
+ * on the field.
+ */
+
+class ViolationFunction
```

Comment:
Hmm, I'd almost change this to something like 
```
using ViolationFunction = std::function<double(Point)>;
```
instead? (Not totally sure if my code is right, but you get the idea).

Reply:
made the change